code/game/gamemodes/cult/runes.dm

```diff
@@ -181,11 +176,52 @@ var/list/sacrificed = list()
 				var/choice = alert(target,"Do you want to join the cult?","Submit to Nar'Sie","Resist","Submit")
 				waiting_for_input[target] = 0
 				if(choice == "Submit") //choosing 'Resist' does nothing of course.
-					cult.add_antagonist(target.mind)
-					converting -= target
-					target.hallucination = 0 //sudden clarity
-					playsound(target, 'sound/effects/bloodcult.ogg', 100, 1)
-
+					if(!target.is_mechanical())
+						cult.add_antagonist(target.mind)
+						converting -= target
+						target.hallucination = 0 //sudden clarity
+						playsound(target, 'sound/effects/bloodcult.ogg', 100, 1)
+					else
+						converting -= target
+						//If we are dealing with a IPC then ask the caster what construct they want
+						var/construct_class = alert(attacker, "Please choose which type of construct you wish to create.",,"Juggernaut","Wraith","Artificer")
+
+						playsound(target, 'sound/effects/bloodcult.ogg', 100, 1)
+
+						//Spawn some remains
+						var/obj/effect/decal/remains/remains = target.species.remains_type //spawns a skeleton based on the species remain type

```

Comment:
Can't this be condensed into `new target.species.remains_type(target.loc)` on one line?